coral-trino/src/main/java/com/linkedin/coral/trino/trino2rel/TrinoCalciteUDFTransformer.java

```diff
@@ -0,0 +1,372 @@
+ * - A vector function g for transforming all operands
+ * - A function h for transforming the result.
+ *
+ * This class will represent g and h as expressions in JSON format as follows:
+ * - Operators: +, -, *, /, and ^
+ * - Operands: source operands and literal values
+ *
+ * There may also be situations where a function in one language can map to more than one functions in the other
+ * language depending on the set of input parameters.
+ * We define a set of matching functions to determine what function name is used.
+ * Currently, there is no use-case more complicated than matching a parameter string to a static regex.
+ *
+ * Example 1:
+ * In Calcite SQL, TRUNCATE(aDouble, numDigitAfterDot) truncates aDouble by removing
+ * any digit from the position numDigitAfterDot after the dot, like truncate(11.45, 0) = 11,
+ * truncate(11.45, 1) = 11.4
+ *
+ * In Trino, TRUNCATE(aDouble) only takes one argument and removes all digits after the dot,
+ * like truncate(11.45) = 11.
+ *
+ * The transformation from Calcite TRUNCATE to Trino TRUNCATE is represented as follows:
+ * 1. Trino name: TRUNCATE
+ *
+ * 2. Operand transformers:
+ * g(b1) = a1 * 10 ^ a2, with JSON format:
+ * [
+ *  { "op":"*",
+ *    "operands":[
+ *      {"input":1}, // input 0 is reserved for result transformer. source inputs start from 1
+ *      { "op":"^",
+ *        "operands":[
+ *          {"value":10},
+ *          {"input":2}]}]}]
+ *
+ * 3. Result transformer:
+ * h(result) = result / 10 ^ a2
+ * { "op":"/",
+ *    "operands":[
+ *      {"input":0}, // input 0 is for result transformer
+ *      { "op":"^",
+ *        "operands":[
+ *          {"value":10},
+ *          {"input":2}]}]}]
+ *
+ *
+ * 4. Operator transformers:
+ * none
+ *
+ * Example 2:
+ * In Calcite, there exists a hive-derived function to decode binary data given a format, DECODE(binary, scheme).
+ * In Trino, there is no generic decoding function that takes a decoding-scheme.
+ * Instead, there exist specific decoding functions that are first-class functions like FROM_UTF8(binary).
+ * Consequently, we would need to know the operands in the function in order to determine the corresponding call.
+ *
+ * The transformation from Calcite DECODE to a Trino equivalent is represented as follows:
+ * 1. Trino name: There is no function name determined at compile time.
+ * null
+ *
+ * 2. Operand transformers: We want to retain column 1 and drop column 2:
+ * [{"input":1}]
+ *
+ * 3. Result transformer: No transformation is performed on output.
+ * null
+ *
+ * 4. Operator transformers: Check the second parameter (scheme) matches 'utf-8' with any casing using Java Regex.
+ * [ {
+ *    "regex" : "^.*(?i)(utf-8).*$",
+ *    "input" : 2,
+ *    "name" : "from_utf8"
+ *   }
+ * ]
+ */
+class TrinoCalciteUDFTransformer {
+  private static final Map<String, SqlOperator> OP_MAP = new HashMap<>();
+
+  // Operators allowed in the transformation
+  static {
+    OP_MAP.put("+", SqlStdOperatorTable.PLUS);
```

Comment:
`OP_MAP` is used to figure out the corresponding SqlOperator for primitive operators within a json transformer. For example we register `"*"` as `SqlStdOperatorTable.MULTIPLY`, so that if we encounter a json transformer like this it knows what the "op" means when doing the transformation:  
```
 [
   { "op":"*",
     "operands":[
       {"input":1}, // input 0 is reserved for result transformer. source inputs start from 1
       { "op":"^",
         "operands":[
           {"value":10},
           {"input":2}]}]}]
```

In contrast, the `TRANSFORMER_MAP` maps a Trino Sql operator names to the corresponding `TrinoCalciteOperatorTransformer`. This map is manually populated.